Add Guide to provider documentation describing how to test out ephemeral resources (#12453)
GoogleCloudPlatform/magic-modules: mmv1/third_party/terraform/website/docs/guides/using_ephemeral_resources.html.markdown

Ephemeral resources are available in Terraform v1.10 and later. For more information, see the [official HashiCorp documentation for Ephemeral Resources](https://developer.hashicorp.com/terraform/language/resources/ephemeral).

To mark the launch of the ephemeral resources feature, the Google Cloud provider has added four ephemeral resources:
- [`google_service_account_access_token`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/ephemeral-resources/service_account_access_token)
- [`google_service_account_id_token`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/ephemeral-resources/service_account_id_token)
- [`google_service_account_jwt`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/ephemeral-resources/service_account_jwt)
- [`google_service_account_key`](https://registry.terraform.io/providers/hashicorp/google/latest/docs/ephemeral-resources/service_account_key)

These are based on existing data sources already in the provider. In future you may wish to update your configurations to use these ephemeral version, as they will allow you to avoid storing tokens and credentials values in your Terraform state.

> ## Using the Google Cloud provider's new ephemeral resources

Author: ```suggestion
## Use the Google Cloud provider's new ephemeral resources
```

> Ephemeral resources are a source of ephemeral data, and they can be referenced in your configuration just like the attributes of resources and data sources. A field, for example in a resource, that references an ephemeral resource must be capable of handling ephemeral data. Due to this, resources in the Google Cloud provider will need to be updated in future to have write-only attributes that are capable of using ephemeral data while not storing those values in the resource's state. 

Author: ```suggestion
Ephemeral resources are a source of ephemeral data, and they can be referenced in your configuration just like the attributes of resources and data sources. However, a field that references an ephemeral resource must be capable of handling ephemeral data. Due to this, resources in the Google Cloud provider will need to be updated so they include write-only attributes that are capable of using ephemeral data while not storing those values in the resource's state. 
```

```
ephemeral.google_service_account_key.mykey: Configuration unknown, deferring...

Terraform used the selected providers to generate the
following execution plan. Resource actions are indicated
with the following symbols:
  + create

Terraform will perform the following actions:

  # google_service_account.myaccount will be created
  + resource "google_service_account" "myaccount" {
    ...
```

During `terrform apply` you will see the ephemeral resource is the final resource to be evaluated, because it depends on the two other resources, and the ephemeral resource is not reflected in the statistics about how many resources were created during the apply action:

```
ephemeral.google_service_account_key.mykey: Opening...
ephemeral.google_service_account_key.mykey: Opening complete after 1s
ephemeral.google_service_account_key.mykey: Closing...
ephemeral.google_service_account_key.mykey: Closing complete after 0s

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

If you run the example using the local backend you can also inspect the state, where you will see that the ephemeral resource is not represented.

> ### Test using ephemeral resources to configure the Google Cloud provider

Author: ```suggestion
### Use ephemeral resources to configure the Google Cloud provider
```

The [documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/ephemeral-resources/service_account_access_token) for the `google_service_account_access_token` ephemeral resource demonstrates how it can be used to configure the provider. Check that ephemeral resource's documentation for details about the IAM permissions required for this example to work:

```hcl
provider "google" {
}

> data "google_client_config" "default" {
>   provider = google
> }

Author: This isn't used in the example

```suggestion
```
